app/ide-desktop/client/tests/localWorkflow.spec.ts

```diff
@@ -148,6 +148,10 @@ test('Local Workflow', async ({ page, app, projectsDir }, testInfo) => {
   expect(projectFiles).toContain('images')
   const images = await fs.readdir(pathModule.join(PROJECT_PATH, 'images'))
   expect(images).toContain('image.png')
+
+  // Ensure that the graph is stable before doing the snapshot. Some nodes might still
+  // be pending at the time the image upload finishes, which can cause the snapshot to
+  // be unstable.
```

Comment:
I assumed before, that nodes cannot become pending after docs change, but if it's not true, then this comment perhaps is not needed. (Or just briefly mention, that pasting image might have turned some nodes pending again).